# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositor.h
# Commit: 9d637c05af7f
# Full Hash: 9d637c05af7f8c39af78b290335ca55994849ee8
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r ea88b243717a -r 9d637c05af7f gfx/webrender_bindings/RenderCompositor.h
--- a/gfx/webrender_bindings/RenderCompositor.h	Fri Nov 06 00:00:47 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositor.h	Fri Nov 06 00:01:36 2020 +0000
@@ -30,6 +30,8 @@
 
 namespace wr {
 
+class RenderCompositorD3D11SWGL;
+
 class RenderCompositor {
  public:
   static UniquePtr<RenderCompositor> Create(
@@ -85,6 +87,10 @@
     return layers::WebRenderCompositor::DRAW;
   }
 
+  virtual RenderCompositorD3D11SWGL* AsRenderCompositorD3D11SWGL() {
+    return nullptr;
+  }
+
   // True if AttachExternalImage supports being used with an external
   // image that maps to a RenderBufferTextureHost
   virtual bool SupportsExternalBufferTextures() const { return false; }